src/features/party/PartiesProvider.tsx

```diff
@@ -103,49 +82,85 @@ const { Provider: RealCurrentPartyProvider, useCtx: useCurrentPartyCtx } = creat
   },
 });
 
+function useAltinnPartyIdCookie() {
+  return useQuery({
+    queryKey: partyQueryKeys.cookie,
+    queryFn: () => getCookieValue(altinnPartyIdCookieName),
+    enabled: true,
+  });
+}
+
+const useSetAltinnPartyIdCookie = () => {
+  const queryClient = useQueryClient();
+  return useMutation({
+    mutationFn: async (partyId: string | number | undefined) => {
+      const value = partyId ?? '';
+      setCookie({ name: altinnPartyIdCookieName, value });
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: partyQueryKeys.cookie });
+    },
+  });
+};
+
+function findCookieParty(partyId: string | null | undefined, parties: IParty[]): IParty | undefined {
+  if (!partyId) {
+    return undefined;
+  }
+
+  return parties.find((party) => {
+    if (party.partyId.toString() === partyId) {
+      return true;
+    }
+
+    if (party.childParties) {
+      return findCookieParty(partyId, party.childParties);
+    }
+  });
+}
+
 const CurrentPartyProvider = ({ children }: PropsWithChildren) => {
+  const {
+    data: allParties,
+    isLoading: isLoadingParties,
+    error: allPartiesError,
+  } = useQuery({
+    queryKey: partyQueryKeys.all,
+    queryFn: fetchAllParties,
+  });
   const validParties = useValidParties();
-  const [sentToMutation, setSentToMutation] = useState<IParty | undefined>(undefined);
-  const { mutateAsync, data: dataFromMutation, error: errorFromMutation } = useSetCurrentPartyMutation();
-  const { data: partyFromQuery, isLoading, error: errorFromQuery } = useCurrentPartyQuery(true);
+
+  const profile = useProfile();
   const [userHasSelectedParty, setUserHasSelectedParty] = useState(false);
+  const { data: altinnPartyIdCookieValue, isLoading: isLoadingCookie, error: cookieError } = useAltinnPartyIdCookie();
+  const { mutateAsync: setAltinnPartyIdCookie, error: mutationError } = useSetAltinnPartyIdCookie();
+
+  if (!validParties?.length) {
+    return <NoValidPartiesError />;
+  }
 
-  if (isLoading) {
-    return <Loader reason='current-party' />;
+  if (isLoadingCookie || isLoadingParties) {
+    return <Loader reason='parties' />;
   }
 
-  const error = errorFromMutation || errorFromQuery;
+  const error = cookieError ?? allPartiesError ?? mutationError;
   if (error) {
     return <DisplayError error={error} />;
   }
 
-  if (!validParties?.length) {
-    return <NoValidPartiesError />;
-  }
+  const cookieParty = findCookieParty(altinnPartyIdCookieValue, allParties ?? []);
 
-  const partyFromMutation = dataFromMutation === 'Party successfully updated' ? sentToMutation : undefined;
-  const currentParty = partyFromMutation ?? partyFromQuery;
-  const currentIsValid = currentParty && validParties?.some((party) => party.partyId === currentParty.partyId);
+  const currentParty = altinnPartyIdCookieValue ? cookieParty : profile?.party;
+  const currentPartyIsValid = currentParty && validParties?.some((party) => party.partyId === currentParty?.partyId);
```

Comment:
This is the part I'm most unsure of. I see that we stop calling the current party API endpoint completely, and just rely on the partylist. Our lack of control and knowledge of the inner workings in the underlying authorization APIs makes this pretty scary to me. Since this will go out as a patch release, don't we risk breaking a lot of apps?

An alternative approach I think we discussed is making `useCurrentParty` always set the cookie based on the instanceOwner param when available. In that case the current party API would function as a simple lookup using the exact same API as we supposedly should have been using all along. Considering all the callers of this hook, wouldn't that also help keep the cookie more updated? Did you consider it or would that not work for some reason?